src/main/scala/firrtl/transforms/TopWiring.scala

```diff
@@ -0,0 +1,247 @@
+    case w: IsDeclaration =>
+      if (sourceList.contains(ComponentName(w.name, currentmodule))) {
+          val (isport, tpe) = w match {
+            case d: DefWire => (false, d.tpe)
+            case d: DefNode => (false, d.value.tpe)
+            case d: DefRegister => (false, d.tpe)
+            case d: Port => (true, d.tpe)
+            case _ => sys.error(s"Cannot wire this type of declaration! ${w.serialize}")
+          }
+          val name = w.name
+          sourceMap.get(currentmodule.name) match {
+            case Some(xs:Seq[(ComponentName, Type, Boolean, InstPath)]) => sourceMap.update(currentmodule.name, xs :+ (ComponentName(w.name,currentmodule), tpe, isport ,Seq[String](w.name)))
+            case None => sourceMap(currentmodule.name) = Seq((ComponentName(w.name,currentmodule), tpe, isport ,Seq[String](w.name)))
+          }
+          //sourceMap(currentmodule.name) +: (ComponentName(w.name,currentmodule), tpe, isport ,Seq[String](currentmodule.name))
+      }
+      w // Return argument unchanged (ok because DefWire has no Statement children)
+    // If not, apply to all children Statement
+    case _ => s map getSourceTypes(sourceList, sourceMap, currentmodule, state)
+  }
+
+
+
+
+  def getSourceTypesPorts(sourceList: Seq[ComponentName], sourceMap: mutable.Map[String, Seq[(ComponentName, Type, Boolean, InstPath)]], currentmodule: ModuleName, state: CircuitState)(s: Port): CircuitState = s match {
+    // If target port, add name and size to to sourceMap
+    case w: IsDeclaration =>
+      if (sourceList.contains(ComponentName(w.name, currentmodule))) {
+          val (isport, tpe) = w match {
+            case d: Port => (true, d.tpe)
+            case _ => sys.error(s"Cannot wire this type of declaration! ${w.serialize}")
+          }
+          val name = w.name
+          sourceMap.get(currentmodule.name) match {
+            case Some(xs:Seq[(ComponentName, Type, Boolean, InstPath)]) => sourceMap.update(currentmodule.name, xs :+ (ComponentName(w.name,currentmodule), tpe, isport ,Seq[String](w.name)))
+            case None => sourceMap(currentmodule.name) = Seq((ComponentName(w.name,currentmodule), tpe, isport ,Seq[String](w.name)))
+          }
+          //sourceMap(currentmodule.name) +: (ComponentName(w.name,currentmodule), tpe, isport ,Seq[String](currentmodule.name))
+      }
+      state // Return argument unchanged (ok because DefWire has no Statement children)
+    // If not, apply to all children Statement
+    case _ => state
+  }
+
+
+
+
+
+
+
+  /** Create a map of Module name to target wires under this module
+    *
+    * These paths are relative but cross module (they refer down through instance hierarchy)
+    */
+  private def getSourcesMap(state: CircuitState): Map[String, Seq[(ComponentName, Type, Boolean, InstPath)]] = {
+    val sSourcesModNames = getSourceModNames(state)
+    val sSourcesNames = getSourceNames(state)
+     
+
+    val cMap = WiringUtils.getChildrenMap(state.circuit).toMap
```

Comment:
`getChildrenMap` should've been deprecated in favor of the DiGraph/InstanceGraph APIs. The only thing still using that is `ClockList` and that's overdue to be upgraded. Sorry, this should've been marked as deprecated long ago... Fixing that in #801.

Reply:
What would be the correct API then? I could not find an exact equivalent in DiGraph/InstanceGraph

Reply:
There won't be an exact equivalent (and I'd have to actually think of what you want here), but the genericity of DiGraph/InstanceGraph should give you all the primitives you need here. It won't be the exact same API, but `InstanceGraph` usually takes care of all of these things. 

ChildrenMap should be the edges for all WDefInstance nodes that share the same module name.